macro/vignettes/pfSI.Rmd

```diff
@@ -0,0 +1,385 @@
+  + GS - gametocyte-blocking vaccination. incomplete by U(0, 1) < gs_protect_pf. Lowers
+    c, the probability to infect mosquitoes.
+  + GS with treatment.
+  + wane PE - (-PE, +S) rate normal(mnPE_PF, vr_PE_Pf). Sets b back to full power.
+  + wane GS - (-GS, +S) rate normal(mnGE_PF, vr_GE_Pf). Sets c back to full power.
+* Bites - These will move to another module, the biting module.
+  + The biting algorithm returns a number of bites for this person.
+  + The biting hazard is incomplete. There is a probability draw that
+    U(0, 1) < b to create move the person to the latent state.
+
+
+## Class of problems
+
+The pfSI described above fits into a particularly restrictive class of models
+because its state space is discrete and entirely Markovian.
+We have examples of simulations we'd like to write that don't fit those
+requirements. Let's look at some examples.
+
+Consider the case of assigning infectious bites to a person.
+We would like to draw the probability that a bite is infectious and the
+queue the event only if it is infectious. The way this is done for a Markov process
+is to specify the distribution of bite times as an incomplete distribution,
+meaning the maximum cumulative probability is the binomial probability that
+the bite is infectious. Maybe that's harder to figure out than thinking of it
+as two draws.
+
+Another case is when a change of state can lead to one of multiple transitions.
+In a Markov process, we would enumerate possible outcome states so that it's clear
+the system can't go to more than one. Then schedule them as competing processes.
+It can be easier to implement this behavior as a multinomial draw followed
+by scheduling the chosen event.
+
+
+# Continuous Time Model
+
+A continuous-time algorithm is a good choice when there will be few events in
+each time step. In this case, most processes are slow relative to a day, so we
+suspect continuous-time is a good choice, even in R where a single time step can 
+be expensive.
+
+We can make a back-of-the-envelope calculation by assuming larger numbers for
+each possible event. If each person gets ten bouts of
+malaria in their lifetime, that's infection, treatment, and back to susceptible, for
+thirty events. If we consider infectious bites, it's either those same infection events
+or we're counting a lot more bites, in the thousands. For a ten year simulation, it's
+3600 days in ten years. We either have events in 30/3600 days or over one per day.
+In the former case, continuous time would likely do better. In the latter case, discrete
+time has an advantage.
+
+
+## Implementation
+
+We can represent the state above as a disease state, treatment state,
+and scheduled bites.
+```{r list_state}
+individual <- list(
+  disease = factor("S", levels = c("S", "I")),
+  treatment = factor("U", levels = c("U", "PE", "GS")),
+  bites = c(0.1, 0.4, 5.7, 7.4)
+  )
+```
+Then we have about twelve transitions. We usually think about these rules
```

Comment:
Is there a reason to not be specific about the number of allowed transitions?

Reply:
thirteen. unlucky. I changed it.